### DISCUSSION TITLE
Question about embedding (RAG) and suggestion about image generation (ComfyUI)

### DISCUSSION THREAD

**Discussion by YarvixPA** [open-webui/open-webui#6426] [General]
Hello.

I am not an expert, I have just been using these LLM models for a few days and I am very interested in having the ability to use them locally and be able to use my documents in conversations.

In the latest version (v0.3.33) I notice an improvement when converting uploaded files to text. The created text is better written and preserves the structure of the file better.

However, it often happens to me that even if I use the file or the created knowledge. When I ask for example: "Give me information about the document" or "Rewrite the information about the document" (not so clear sentences related to the context of the document) and similar things. It does not always use the context of the documentation.

Embedding models that I have seen that work best (taking into account that they are multilingual and have good dimensions in the embedding space):

- BGE-M3 (1024)
- Granite 3 (1B MoE: 1024, 3B MoE: 1536, 2B Dense: 2048, 8B Dense: 4096)
- jina-embeddings-v3 (32, 64, 128, 256, 512, 768, 1024)

Reranker models (In these models I have also opted for the multilingual models):

- BAAI/bge-reranker-v2-m3
- jina-reranker-v2-base-multilingual

Why have I used these multilingual models? Because I consider that I will not always have or use documents that are in only one language.

Something I did notice in the previous version is that the created text was poorly formatted when created and I don't know if that was part of the problem.

I imagine there are ways to continue improving the results when performing RAG. I would like to receive information on how I could improve this.

-----------------

Also another thing I would like to comment on is about the generation of images...

I would like and find it more practical to use commands to call the generation of images. Something similar to how "Knowledge" is called. This will allow the same prompt or user message to be the one sent for image generation... something like it is done in Midjourney would be useful

/ or @ "Generate image" + prompt

In my case I use ComfyUI... it would be good to be able to create preset lists (something like when adding more Ollama URLs):

Aspect ratio - Resolution
1:1 - 1024x1024
16:9 - .....

and then in the prompt it would be defined as it is done in Midjourney for example --ar:1:1

**Comment by ovizii** [comment]
Can you share some feedback on how much RAM those embedders and rerankers consume?

If only using the default sentence-transformers/all-MiniLM-L6-v2 I am below 2 GB of RAM, if I switch to jinaai/jina-embeddings-v3 and add a simple and plain document RMA usage goes up to 14 GB. 

I haven't even dared test rerankers, hoping for some feedback from you. I also have the use case that I switch between documents in 3 different languages hence I assume I'd need a multilanguage embedder and reranker?
